client/src/js/components/bhPayrollPeriodSelect.js

```diff
@@ -0,0 +1,63 @@
+angular.module('bhima.components')
+  .component('bhPayrollPeriodSelect', {
+    templateUrl : 'modules/templates/bhPayrollPeriodSelect.tmpl.html',
+    controller  : PayrollPeriodSelectController,
+    transclude  : true,
+    bindings    : {
+      payrollConfigurationId : '<',
+      onSelectCallback : '&',
+      required         : '<?',
+      label            : '@?',
+      name             : '@?',
+      validationTrigger :  '<?',
+    },
+  });
+
+PayrollPeriodSelectController.$inject = [
+  'PayrollConfigurationService', '$timeout', '$scope', 'NotifyService',
+];
+
+/**
+ * Payroll Period Select Controller
+ */
+function PayrollPeriodSelectController(PayrollConfigs, $timeout, $scope, Notify) {
+  const $ctrl = this;
+
+  // fired at the beginning of the Payroll Configuration select
+  $ctrl.$onInit = function $onInit() {
+    // translated label for the form input
+    $ctrl.label = $ctrl.label || 'FORM.LABELS.PERIOD_PAYMENT';
+
+    // fired when an Payroll Period configuration has been selected
+    $ctrl.onSelectCallback = $ctrl.onSelectCallback;
```

Comment:
Just FYI - this doesn't do anything.  I've seen it in a couple pull requests, but it is basically like writing:
```js
var x = 2;
x = x;  // not necessary
```